src/tools/dev/scripts/bv_support/bv_llvm.sh

```diff
@@ -29,14 +29,14 @@ function bv_llvm_info
     export BV_LLVM_FILE=${BV_LLVM_FILE:-"llvm-${BV_LLVM_VERSION}.src.tar.xz"}
     export BV_LLVM_URL=${BV_LLVM_URL:-"http://releases.llvm.org/${BV_LLVM_VERSION}/"}
     export BV_LLVM_BUILD_DIR=${BV_LLVM_BUILD_DIR:-"llvm-${BV_LLVM_VERSION}.src"}
-    export LLVM_MD5_CHECKSUM="c88c98709300ce2c285391f387fecce0"
-    export LLVM_SHA256_CHECKSUM="b6d6c324f9c71494c0ccaf3dac1f16236d970002b42bb24a6c9e1634f7d0f4e2"
+    export BV_LLVM_MD5_CHECKSUM="c88c98709300ce2c285391f387fecce0"
+    export BV_LLVM_SHA256_CHECKSUM="b6d6c324f9c71494c0ccaf3dac1f16236d970002b42bb24a6c9e1634f7d0f4e2"
 
     export BV_CLANG_URL=${BV_LLVM_URL}
     export BV_CLANG_FILE="cfe-${BV_LLVM_VERSION}.src.tar.xz"
     export BV_CLANG_BUILD_DIR="cfe-${BV_LLVM_VERSION}.src"
-    export CLANG_MD5_CHECKSUM="4e419bd4e3b55aa06d872320f754bd85"
-    export CLANG_SHA256_CHECKSUM="7c243f1485bddfdfedada3cd402ff4792ea82362ff91fbdac2dae67c6026b667"
+    export BV_CLANG_MD5_CHECKSUM="4e419bd4e3b55aa06d872320f754bd85"
+    export BV_CLANG_SHA256_CHECKSUM="7c243f1485bddfdfedada3cd402ff4792ea82362ff91fbdac2dae67c6026b667"
```

Comment:
Why is 'BV_' prefix required for checksums  here? Not all of our bv modules use the prefix with their checksums. Actually, other modules don't use the 'BV_' prefix for any of the vars in their _info method.
I guess though, if the prefix works and the checksums are used/checked, then adding the prefix for the checksum vars makes them consistent with the other vars in this method.

Reply:
With the names as they were, the checksum variables weren't found and the checksum wasn't checked.

In the function verify_checksum_by_lookup, it looks for all environment variables of the form XXX_FILE whose value is the name of the file it downloaded. In this case it finds BV_CLANG_FILE. It then takes the part of the name preceding "_FILE", which is "BV_CLANG". It then takes that prefix and combines it with "_MD5_CHECKSUM", "_SHA256_CHECKSUM" and "_SHA512_CHECKSUM" to form the checksum variable names. As the names were, it couldn't find the CHECKSUM variable names. There are multiple solutions to this. I assumed you wanted to have it include "BV" in the FILE variable name, possibly to avoid name space issues, so I changed the checksum variable names to match.

Reply:
Makes sense. Thanks for the clarification on how the checksums var names are looked up.